gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnAppLauncher.java

```diff
@@ -885,10 +884,11 @@ private AbstractTokenRefresher buildTokenRefreshManager() throws IOException {
 
   @VisibleForTesting
   void cleanUpAppWorkDirectory(ApplicationId applicationId) throws IOException {
-    Path appWorkDir = GobblinClusterUtils.getAppWorkDirPathFromConfig(this.config, this.fs, this.applicationName, applicationId.toString());
-    if (this.fs.exists(appWorkDir)) {
+    FileSystem fs = GobblinClusterUtils.buildNewInstanceFileSystem(this.config, this.yarnConfiguration);
```

Comment:
let's document that we found the other FS instance was closed by the YARN app and couldn't be reused!

Reply:
what do you think here?